python/cudf/cudf/core/udf/pipeline.py

```diff
@@ -16,7 +18,49 @@
 precompiled: cachetools.LRUCache = cachetools.LRUCache(maxsize=32)
 
 
-def get_frame_row_type(fr):
+class _FrameJitMetadata:
```

Comment:
With the degree to which we've simplified this class, I'm not even sure it needs to exist. `supported_dtypes` is currently only being used in two places, and in one of those if I'm not mistaken [you're basically reconstructing `supported_cols`](https://github.com/rapidsai/cudf/pull/9359/files#diff-5585da0e33fdf9081657432745c8a9d096d0b6d5eba1794d0bbd629661e61a0cR262). If that's correct, then I think in [the other place that it's used](https://github.com/airspeed-velocity/asv/issues/567#issuecomment-960347583) you could just construct it on the fly as `{name: col.dtype for name, col in supported_cols.items()}`. At that point there are only two attributes store in this class and you could probably just construct them in a function rather than with a separate class to hold them.

Reply:
(I secretly hoped this might happen but I didn’t want to request it!)

Reply:
Heard loud and clear- this has been changed to use a couple of functions instead. We will see if the code develops in a way that justifies reintroducing it at some point.